modules/claims_api/lib/custom_error.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class CustomError
+    def initialize(error, claim, method)
+      @error = error
+      @claim = claim
+      @method = method
+    end
+
+    def build_error
+      custom_error = if @error == Faraday::ConnectionFailed || @error == Faraday::ParsingError
+                       { 'messages' => [{ 'key' => 'ServiceException', 'detail' => 'test detail', status: '500' }] }
+
+                     elsif @error.is_a?(::Common::Exceptions::BackendServiceException) # missing bracket on form_data
+                       { 'messages' => [{ 'key' => 'BackendException',
+                                          'detail' => 'Backend exception', status: '500' }] }
+
+                     elsif @error.is_a?(StandardError) # when client_key is blank
+                       { 'messages' => [{ 'key' => 'Client error', 'detail' => 'client exception', status: '400' }] }
+
+                     else
+                       { 'messages' => [{ 'key' => 'Unknown error', 'detail' => 'unknown error', status: '500' }] }
+
+                     end
+      update_claim(custom_error)
+      log_outcome_for_claims_api(custom_error)
+      raise EVSS::DisabilityCompensationForm::ServiceException, custom_error
```

Comment:
raising this exception puts it in a terminal state every time. we're wanting to retry specific errors like a 401 status or something that _could_ succeed. When the Client Key is blank you should receive a 401 status from the EVSS Docker Container - an example of a retry-able event. Obviously the Client Key shouldn't be blank, but if we have a something like an expired token we should continue attempting to process the claim. It would hopefully resolve on the next attempt. 

The original code with the `else` would have worked, we just can't raise an object, so we would want to relay the error here forward which may break the message parsing again, but as mention, this will be "living" logic we'll need to update as we find new exceptions from the Docker code.